src/components/componentDiff/ComponentDiffTab.js

```diff
@@ -219,6 +221,65 @@ const ComponentVersionDiff = ({
     rightVersion: rightVersion,
   })
 
+  const HandleExportDiff = () => {
+    const addedResources = []
+    const removedResources = []
+
+    diff.components_changed?.forEach(component => {
+
+      const formatResourceName = (resource, component) => {
+        if (resource.type === 'ociImage') {
+          return `[Docker] ${component.name.split('/', 1)[0]}/${resource.name}`
+        }
+        return `${component.name.split('/', 1)[0]}/${resource.name}`
+      }
+
+      component.right.resources?.added?.forEach(resource => {
+        addedResources.push(formatResourceForYaml(component.right, resource, formatResourceName))
+      })
+
+      component.right.resources?.removed?.forEach(resource => {
+        removedResources.push(formatResourceForYaml(component.right, resource, formatResourceName))
+      })
+
+
+      component.right.resources?.changed?.forEach(change => {
+        addedResources.push(formatResourceForYaml(component.right, change.to, formatResourceName))
+        removedResources.push(formatResourceForYaml(component.right, change.from, formatResourceName))
+      })
+
+    })
+
+    downloadYaml(Yaml.dump(addedResources, { lineWidth: 1000 }), 'resource-diff-added.yaml')
+    downloadYaml(Yaml.dump(removedResources, { lineWidth: 1000 }), 'resource-diff-removed.yaml')
+  }
+
+  const formatResourceForYaml = (component, resource, formatResourceName) => {
```

Comment:
The function itself does not have anything to do with a yaml file, instead the resource is serialised to an expected output format. So, it'd suggest to express this in the function name as well, e.g. `serialise_resource`.